packages/react/src/time-picker/utils.test.ts

```diff
@@ -0,0 +1,693 @@
+import {
+	clampNumber,
+	filterOptions,
+	formatValue,
+	generateOptions,
+	parseTime,
+} from './utils';
+
+describe('filterOptions', () => {
+	describe('Filters options correctly', () => {
+		const options = [
+			/* 0 */ { label: '00:00', value: '00:00' },
+			/* 1 */ { label: '12:00 am', value: '00:00' },
+			/* 2 */ { label: '9:00', value: '09:00' },
+			/* 3 */ { label: '9:00 am', value: '09:00' },
+			/* 4 */ { label: '09:00 am', value: '09:00' },
+			/* 5 */ { label: '12:00', value: '12:00' },
+			/* 6 */ { label: '12:00 pm', value: '12:00' },
+			/* 7 */ { label: '15:50', value: '15:50' },
+			/* 8 */ { label: '3:50 pm', value: '15:50' },
+			/* 9 */ { label: '03:50 pm', value: '15:50' },
+			/* 10 */ { label: '21:00', value: '21:00' },
+			/* 11 */ { label: '9:00 pm', value: '21:00' },
+			/* 12 */ { label: '09:00 pm', value: '21:00' },
+			/* 13 */ { label: '23:59', value: '23:59' },
+			/* 14 */ { label: '11:59 pm', value: '23:59' },
+			/* 15 */ { label: '12:43 am', value: '00:43' },
+			/* 16 */ { label: '12:43 pm', value: '12:43' },
+		];
+
+		test('12', () => {
+			expect(filterOptions(options, '12')).toEqual([
+				options[0],
+				options[1],
+				options[5],
+				options[6],
+				options[15],
+				options[16],
+			]);
+		});
+		test("12 o'clock", () => {
+			expect(filterOptions(options, "12 o'clock")).toEqual([
+				options[0],
+				options[1],
+			]);
+		});
+		test('12am', () => {
+			expect(filterOptions(options, '12am')).toEqual([options[0], options[1]]);
+		});
+		test('12 am', () => {
+			expect(filterOptions(options, '12 am')).toEqual([options[0], options[1]]);
+		});
+		test('12pm', () => {
+			expect(filterOptions(options, '12pm')).toEqual([options[5], options[6]]);
+		});
+		test('12 pm', () => {
+			expect(filterOptions(options, '12 pm')).toEqual([options[5], options[6]]);
+		});
+		test('1200', () => {
+			expect(filterOptions(options, '1200')).toEqual([options[5], options[6]]);
+		});
+		test('12:00', () => {
+			expect(filterOptions(options, '12:00')).toEqual([
+				options[1],
+				options[5],
+				options[6],
+			]);
+		});
+		test('24', () => {
+			expect(filterOptions(options, '24')).toEqual([
+				options[0],
+				options[1],
+				options[15],
+				options[16],
```

Comment:
If I typed in 24, I wouldn't expect to see `12:... pm`.

Reply:
24 only returns 16 (12:43pm) here because of the 24 in 1**2:4**3. I'll update that to make it clearer